src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -116,32 +116,47 @@ object Parser {
       }
     }
 
-    def isErrorFree(parserResults: Vector[Either[Vector[ParserError], PProgram]]): Either[Vector[ParserError], Vector[PProgram]] = {
-      val (errors, programs) = parserResults.partitionMap(identity)
-      if (errors.isEmpty) Right(programs) else Left(errors.flatten)
+    def isErrorFree(parserResults: Vector[Either[Vector[ParserError], (String, PProgram)]]): Either[Vector[ParserError], Vector[(String, PProgram)]] = {
+      val (errors, pkgIdProgramMap) = parserResults.partitionMap({
+        case Left(value) => Left(value)
+        case Right(value) => Right(value)
+      })
+      if (errors.isEmpty) Right(pkgIdProgramMap) else Left(errors.flatten)
     }
 
-    def samePackage(programs: Vector[PProgram]): Either[Vector[ParserError], Vector[PProgram]] = {
-      require(programs.nonEmpty)
-      val pkgName = programs.head.packageClause.id.name
-      val differingPkgNameMsgs = programs.flatMap(p =>
+    def samePackage(pkgIdProgramTuples: Vector[(String, PProgram)]): Either[Vector[ParserError], Vector[(String, PProgram)]] = {
+      require(pkgIdProgramTuples.nonEmpty)
+      val expectedPkgId = pkgIdProgramTuples.head._1
+      val differingPkgIdMsgs = pkgIdProgramTuples.flatMap({ case (pkgId, p) =>
```

Comment:
`p` seems to be unused

Reply:
should be, it is used to get the package clause one line later